libs/auth-api-lib/src/lib/entities/dto/delegation.dto.ts

```diff
@@ -0,0 +1,33 @@
+import { ApiProperty } from '@nestjs/swagger'
+import {
+  IsBoolean,
+  IsString,
+  IsOptional,
+  IsNumber,
+  IsDateString,
+} from 'class-validator'
+// For use with deleting Inserting and updating Delegation
+export class DelegationDTO {
+  @IsString()
+  @ApiProperty()
+  fromNationalId!: string
+  @IsString()
+  @ApiProperty()
+  fromDisplayName!: string
+  @IsBoolean()
+  @ApiProperty()
+  isFromCompany!: boolean
+  @IsString()
+  @ApiProperty()
+  toNationalId!: string
```

Comment:
This should always be the CurrentUser. Otherwise, I can create a delegation for anyone/anything.

Reply:
fromNationalId should always be the CurrentUser, right?

Reply:
Arrg 🤯😂

So the delegation goes "from" a company "to" a user? Then I log in as that user, and query all delegations going "to" that user, to show in the delegation dropdown.

Makes sense. And then "fromNationalId should always be the CurrentUser, but passing in "fromDisplayName" is a bit weird. "I'm giving you access to me, but I want me to have a custom display name in your dropdown". I imagined the use case the other way around, me adding an alias for the list of people that have access to me, in findAllFrom.

If we want it for caching reasons, maybe it's better to fetch it from the ID token when creating the delegation. But as I said elsewhere, we'll need to solve this also somehow when querying findAllFrom.